Vagabonds-Labs/cofiblocks
# refactor: Added search bar using jotai Atoms no unnecesary files

## Description

This PR implements the product search bar UI, fetching product data from Mocked data and displaying them in the product list

* Product search bar UI: The product search bar allows you to search and display products that match a search parameter.
* tRPC API: Using the productRouter I have added a new method to view products by name.
* Jotai atoms: Using jotai Atoms to create global react states

## Checklist

- [ ] The title of the PR is clear and descriptive.
- [ ] Code follows project conventions (Biome formatting, linting).
- [ ] Commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style.

## Type of Change


## Testing

### Manual Testing


1. Run the app
2. Search a product for region, for example: "Alajuela"
3. Result will appear
4. Clear the search

If the search parameter does not match, the result will show a "No results" image.

### Automated Tests
- [ ] E2E tests added or updated

## Additional Notes
